addons/mail/static/src/messaging/entity/entity/entity.js

```diff
@@ -153,118 +153,45 @@ function EntityFactory() {
-                            relationName,
-                            relationValue,
-                        });
-                        break;
-                }
-            }
+        update(data = {}) {
+            this.env.entityManager.update(this, data);
         }
 
+        //--------------------------------------------------------------------------
+        // Private
+        //--------------------------------------------------------------------------
+
         /**
-         * Update relation on this instance entity with provided data.
-         *
-         * Any relation, regardless of their type, are processed as removal
-         * of provided value.
-         *
-         * Key should be named after relation names, and each values should be
-         * either local id(s) or entity(ies) that respect the definition of
-         * this relation. Special value `null` can be used to remove all
-         * relations of provided relation name.
-         *
+         * @static
+         * @private
          * @param {Object} data
+         * @param {any} data.id
+         * @return {function}
          */
-        unlink(data) {
-            if (!this.constructor.get(this)) {
-                // Entity has already been deleted.
-                // (e.g. unlinking one of its reverse relation was causal)
-                return;
-            }
-            for (const [relationName, relationValue] of Object.entries(data)) {
-                const relation = this.constructor.fields[relationName];
-                switch (relation.type) {
-                    case 'one2one':
-                        this.constructor.__unlinkSingleOne2One(this, { relationName });
-                        break;
-                    case 'one2many':
-                        this.constructor.__unlinkSingleOne2Many(this, { relationName, relationValue });
-                        break;
-                    case 'many2one':
-                        this.constructor.__unlinkSingleMany2One(this, { relationName });
-                        break;
-                    case 'many2many':
-                        this.constructor.__unlinkSingleMany2Many(this, { relationName, relationValue });
-                        break;
-                }
-            }
+        static _findFunctionFromData(data) {
+            return entity => entity.id === data.id;
         }
 
         /**
-         * Update this instance entity with provided data.
-         *
-         * Provided data is used to update this instance entity for the 1st time.
-         * These data should be explicitly handled in `_update()` method,
-         * otherwise they are ignored.
-         *
-         * @param {Object} [data={}]
+         * @private
+         * @returns {mail.messaging.entity.Messaging}
          */
-        update(data = {}) {
-            this._update(data);
+        _computeMessaging() {
+            return [['link', this.isMessaging ? this : this.env.messaging]];
```

Comment:
This compute is just a default value, so it can be done a create (not compute), and done only where needed (so thread)